src/ng/directive/ngOptions.js

```diff
@@ -489,6 +495,9 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
 
         selectCtrl.readValue = function readNgOptionsValue() {
 
+          // Allow a default value to be substituted for any options with empty values
+          if (selectElement.val() === '') return $parse(attr.ngEmptyValue)(scope);
```

Comment:
The expression $parse should be moved out the readValue fn. It only has to be done once.
So above the function ,do something like `var emptyValueParseFn = 'ngEmptyValue' in attr && attr.ngEmptyValue.length > 0 && $parse(attr.ngEmptyValue);`

and inside the read, do 
`if (selectElement.val() === '' && emptyValueParseFn) return emptyValueParseFn(scope);`

This also ensures we don't use the emptyValueParseFn when the attribute is not defined or if it is empty.
It's a bit odd that this isn't caught by the tests. We should enhance one of the exisiting tests to make sure that the model value is `null` when no ng-empty-value attribute is found.